### PR TITLE
Replace custom attributes with valid w3c attributes

### PR DESCRIPTION
**Pull request by core23**
According to w3c custom attributes should start with `data-*`
- [ ] Update https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle
- [ ] Update https://github.com/sonata-project/SonataDoctrineORMAdminBundle
- [ ] Update https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle
- [ ] Update https://github.com/sonata-project/SonataPropelAdminBundle

### CODE REVIEW

**Comment by OskarStark** [comment]
@rande @Soullivaneuh can you please review this change?

Thank you

File: Resources/views/CRUD/base_list_field.html.twig [hunk 1]
```diff
@@ -9,7 +9,7 @@ file that was distributed with this source code.
 
 #}
 
-<td class="sonata-ba-list-field sonata-ba-list-field-{{ field_description.type }}" objectId="{{ admin.id(object) }}"{% if field_description.options.row_align is defined %} style="text-align:{{ field_description.options.row_align }}"{% endif %}>
+<td class="sonata-ba-list-field sonata-ba-list-field-{{ field_description.type }}" data-objectId="{{ admin.id(object) }}"{% if field_description.options.row_align is defined %} style="text-align:{{ field_description.options.row_align }}"{% endif %}>
```
Comment (Soullivaneuh) [inline Resources/views/CRUD/base_list_field.html.twig]: Maybe take profit to rename `data-objectId` to `data-object-id`.

**Comment by Soullivaneuh** [comment]
@core23 This could be BC break for dev using custom js with those attributes.

What do you think @rande?

**Comment by core23** [comment]
@rande Can you please review?

**Comment by core23** [comment]
Ping @rande

**Comment by rande** [comment]
I think this is a major BC break. we can't merge this as it will break custom code and maybe related sonata bundles. It is a mistake that we need to keep.

**Comment by core23** [comment]
@rande Why did you close this ticket? As it is a bug, it should get merged with next major release.

**Comment by rande** [comment]
@core23 not sure it is a bug, not respecting a standard is not a bug, it is a mistake. So respecting this recommendation might generate a lot of works and a complex update path.

**Comment by core23** [comment]
But not respecting standards could cause some bugs, especially when depending on other libraries as jQuery. 

As you can see there are only a few files affected for this bug.

**Comment by core23** [comment]
@rande Maybe we could then schedule this for the next major release?

**Comment by rande** [comment]
@core23 yes

**Comment by Soullivaneuh** [comment]
Please fill your PR of missing parts indicated by the "Needs" labels.

Feel free to ask if you want some precision of what is needed.

**Comment by core23** [comment]
Added missing Changelog and Upgrade notes

File: UPGRADE-4.0.md [hunk 2]
```diff
@@ -8,3 +8,9 @@ If you have implemented a custom admin, you must adapt the signature of the foll
 ## AdminExtension
 If you have implemented a custom admin extension, you must adapt the signature of the following new methods to match the one in `AdminExtensionInterface` again:
  * `configureActionButtons`
+
+## Templates
+If you extended the ``base_list.html.twig`` (or any child) template, you should replace the following attributes:
+  * ``objectId`` with ``data-object-id``
+  * ``filter-target`` with ``data-filter-target``
+  * ``sonata-filter`` with ``data-sonata-filter``
```
Comment (OskarStark) [inline UPGRADE-4.0.md]: data-filter-container also

**Comment by core23** [comment]
@ju1ius Can have a look at this and review?

**Comment by ju1ius** [comment]
@core23 This is a very good idea.
However keep in mind that the sonata UI is currently spread over (too :trollface:) many bundles (core, admin, orm, odm, phpcr, propel, media, page, user...). The first thing that comes to mind is that the `objectId` attribute change **will** break the form association popup code. So you'll have to make related PRs for at least the 4 persistence bundles. [Example](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/search?utf8=%E2%9C%93&q=objectId&type=Code).
That's why I want to take as much code as possible back to the admin-bundle mothership.

File: UPGRADE-4.0.md [hunk 3]
```diff
@@ -8,3 +8,10 @@ If you have implemented a custom admin, you must adapt the signature of the foll
 ## AdminExtension
 If you have implemented a custom admin extension, you must adapt the signature of the following new methods to match the one in `AdminExtensionInterface` again:
  * `configureActionButtons`
+
+## Templates
+If you extended the ``base_list.html.twig`` (or any child) template, you should replace the following attributes:
```
Comment (Soullivaneuh) [inline UPGRADE-4.0.md]: Only one backtick please.

Same for lines below.

**Comment by Soullivaneuh** [comment]
@sonata-project/contributors does anyone manually tested this one?

@core23 About upgrade note, I think something should be told about the `Admin.js` file. Some user could be using a custom one.

**Comment by core23** [comment]
Thanks for your feedback @ju1ius 

I will wait for you changes on the different admin bundles, so we have less pain with merging.

**Comment by core23** [comment]
Closing this for now. We should refactor the whole JavaScript stack

**Pull request closed without merging**